streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -384,6 +387,13 @@ public String queryableStoreName() {
     private SuppressedInternal<K> buildSuppress(final Suppressed<? super K> suppress) {
         if (suppress instanceof FinalResultsSuppressionBuilder) {
             final long grace = findAndVerifyWindowGrace(streamsGraphNode);
+            if (grace > Duration.ofHours(1).toMillis()) {
+                LOG.warn("Adding suppression to a windowed stream with a suspiciously long grace period of [{}]. " +
+                             "Suppressed results are delayed until the window closes and the grace period elapses. " +
+                             "If you wish to see fresher results, you can set it on the window definition using " +
+                             "'.grace(gracePeriodMs)'.",
+                         Duration.ofMillis(grace).toString());
```

Comment:
maybe display the `grace` period in minutes so users can immediately grok the time frame without needing to convert to minutes themselves?

Reply:
This line actually prints in ISP-8601 format, which breaks down hours, minutes, seconds, and sub-seconds.

https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#toString--

I had the same thought.